File: easy/0346_moving_average_from_data_stream/solution.py
from collections import deque


class MovingAverage:
    def __init__(self, size: int):
        self.__queue = deque()
        self.__size = size

        self.__sum = 0

    def next(self, val: int) -> float:
        if len(self.__queue) == self.__size:
            self.__sum -= self.__queue.popleft()

        self.__queue.append(val)
        self.__sum += val

        return self.__sum / len(self.__queue)

File: easy/0346_moving_average_from_data_stream/solution.cpp
using namespace std;

#include <queue>

class MovingAverage {
private:
  queue<int> q;
  int size;
  double sum;

public:
  MovingAverage(int size) : size(size), sum(0) {}

  double next(int val) {
    if (q.size() == size) {
      sum -= q.front();
      q.pop();
    }

    q.push(val);
    sum += val;

    return sum / q.size();
  }
};
